### PR TITLE
fix(test): check minor version during update-24.04

### PR DESCRIPTION
**Pull request by hamzabessa**
## Description

Enable you to check in the wizard (at step 3 ) that the start and finish versions are correctly quoted in the interface.
checking that the Login page also contains the correct target version.
**Fixes** # MON-138523

## Type of change

- [x] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [ ] 22.10.x
- [ ] 23.04.x
- [ ] 23.10.x
- [x] 24.04.x
- [ ] master

<h2> How this pull request can be tested ? </h2>

Please describe the **procedure** to verify that the goal of the PR is matched. Provide clear instructions so that it can be **correctly tested**.

Any **relevant details** of the configuration to perform the test should be added.

#### Community contributors & Centreon team

- [x] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: centreon/tests/e2e/features/Platform-upgrade-update/01-platform-update/index.ts [hunk 1]
```diff
@@ -114,7 +114,11 @@ Given(
           } else {
             switch (version_from_expression) {
               case 'last stable':
-                minor_version_index = stable_minor_versions.length - 1;
+              const stableMinorVersion = Cypress.env('stableMinorVersion');
+               minor_version_index = stable_minor_versions.length - 1;
```
Comment (kduret) [inline centreon/tests/e2e/features/Platform-upgrade-update/01-platform-update/index.ts:118]: linting issue.
Please run eslint to check coding style issues

File: centreon/tests/e2e/features/Platform-upgrade-update/common.ts [hunk 2]
```diff
@@ -368,7 +380,10 @@ Then(
         template: 'serviceTemplate1'
       })
       .applyPollerConfiguration();
-
+    cy.visit(`${Cypress.config().baseUrl}`);
```
Comment (kduret) [inline centreon/tests/e2e/features/Platform-upgrade-update/common.ts:383]:
```suggestion
    cy.visit('/');
```

File: centreon/tests/e2e/features/Platform-upgrade-update/01-platform-update/index.ts [hunk 3]
```diff
@@ -114,7 +114,11 @@ Given(
...
+                if (stable_minor_versions[minor_version_index] === stableMinorVersion) {
```
Comment (kduret) [inline centreon/tests/e2e/features/Platform-upgrade-update/01-platform-update/index.ts:119]: `stableMinorVersion` seems to be an array, this cannot work

File: centreon/tests/e2e/features/Platform-upgrade-update/common.ts [hunk 4]
```diff
@@ -266,6 +269,7 @@ const checkPlatformVersion = (platformVersion: string): Cypress.Chainable => {
       name: 'web'
     })
     .then(({ output }): Cypress.Chainable<null> | null => {
+
```
Comment (kduret) [inline centreon/tests/e2e/features/Platform-upgrade-update/common.ts:272]:
```suggestion
```

File: centreon/tests/e2e/features/Platform-upgrade-update/common.ts [hunk 5]
```diff
@@ -83,6 +83,9 @@ const getCentreonStableMinorVersions = (
         name: 'web'
       });
     }
+    const stableMinorVersion = [...new Set(stableVersions)].sort((a, b) => a - b).pop();
+    cy.log('stableMinorVersion',stableMinorVersion)
+    Cypress.env('stableMinorVersion', stableMinorVersion);
```
Comment (kduret) [inline centreon/tests/e2e/features/Platform-upgrade-update/common.ts:88]:
```suggestion
    const lastStableMinorVersion = [...new Set(stableVersions)].sort((a, b) => a - b).pop();
    cy.log('lastStableMinorVersion ', lastStableMinorVersion );
    Cypress.env('lastStableMinorVersion ', lastStableMinorVersion );
```

File: centreon/tests/e2e/features/Platform-upgrade-update/01-platform-update/index.ts [hunk 6]
```diff
@@ -114,7 +114,11 @@ Given(
           } else {
             switch (version_from_expression) {
               case 'last stable':
-                minor_version_index = stable_minor_versions.length - 1;
+              const stableMinorVersion = Cypress.env('stableMinorVersion');
```
Comment (kduret) [inline centreon/tests/e2e/features/Platform-upgrade-update/01-platform-update/index.ts:117]:
```suggestion
```

File: centreon/tests/e2e/features/Platform-upgrade-update/01-platform-update/index.ts (hunk 3 above)
Comment (kduret) [inline centreon/tests/e2e/features/Platform-upgrade-update/01-platform-update/index.ts:119]:
```suggestion
                if (stable_minor_versions[minor_version_index] === Cypress.env('lastStableMinorVersion')) {
```

**Review by kduret — APPROVED** [review]

**Review by MedWassCh — DISMISSED** [review]

**Review by MedWassCh — APPROVED** [review]

**Pull request merged**